superset/db_engine_specs/base.py

```diff
@@ -422,10 +435,13 @@ def _allowed_file(filename: str) -> bool:
         }
         df = cls.csv_to_df(**csv_to_df_kwargs)
 
+        sqla_uri = form.con.data.sqlalchemy_uri_decrypted
         df_to_sql_kwargs = {
             "df": df,
             "name": form.name.data,
-            "con": create_engine(form.con.data.sqlalchemy_uri_decrypted, echo=False),
+            "con": create_engine(
+                sqla_uri, echo=False, paramstyle=cls.dialect_paramstyle(sqla_uri)
```

Comment:
I'm wondering if this will subtly break CSV upload for non-Postgres databases. I wonder if we should retain the previous behavior in `db_engine_specs/base.py` and override the `dialect_paramstyle` function for the Postgres driver?